tests/featureWriters/kernFeatureWriter_test.py

```diff
@@ -1840,6 +1840,76 @@ def test_kern_hira_kana_hrkt(FontClass):
     )
 
 
+# flake8: noqa: B950
+def test_redefining_class_defs(FontClass, datadir):
+    """Test that class definitions aren't overwritten by either the same content
+    or silently redefined.
+
+    In this example, the group @kern1.Telu.TEL_ssatelugu.alt_L would be defined
+    in the first pass dealing with base kern pairs, but would be redefined in
+    the mark pair pass with different glyphs (issue #702). Instead, reuse
+    whatever group name was chosen for a globally unique sequence of glyph
+    names.
+    """
+
+    ufo = FontClass(datadir / "RedefinedClassDef.ufo")
+    newFeatures = KernFeatureWriterTest.writeFeatures(ufo)
+
+    assert dedent(str(newFeatures)) == dedent(
+        """\
+        @kern1.Default.TEL_ssatelugu.alt_L = [ss-telugu.alt sse-telugu.alt ssee-telugu.alt ssi-telugu.alt ssii-telugu.alt];
+        @kern1.Default.TEL_ssatelugu_L = [ss-telugu sse-telugu ssee-telugu ssi-telugu ssii-telugu];
+        @kern1.Telu.TEL_ssatelugu.alt_L = [ssa-telugu.alt];
+        @kern1.Telu.TEL_ssatelugu_L = [ssa-telugu];
+        @kern2.Default.TEL_batelugu.below_R = [ba-telugu.below bha-telugu.below];
+        @kern2.Default.TEL_comma_R = [comma.loclTELU semicolon.loclTELU];
+        @kern2.Default.TEL_yatelugu.below_R = [ya-telugu.below];
+        @kern2.Telu.TEL_rVocalicMatratelugu_R = [rVocalicMatra-telugu rrVocalicMatra-telugu];
+
+        lookup kern_Telu {
+            lookupflag IgnoreMarks;
+            pos @kern1.Telu.TEL_ssatelugu_L @kern2.Default.TEL_comma_R 100;
+            pos @kern1.Telu.TEL_ssatelugu.alt_L @kern2.Default.TEL_batelugu.below_R 180;
+            pos @kern1.Telu.TEL_ssatelugu.alt_L @kern2.Default.TEL_comma_R 90;
+            pos @kern1.Telu.TEL_ssatelugu.alt_L @kern2.Default.TEL_yatelugu.below_R 20;
+        } kern_Telu;
+
+        lookup kern_Telu_marks {
+            pos @kern1.Default.TEL_ssatelugu.alt_L @kern2.Telu.TEL_rVocalicMatratelugu_R 180;
+            pos @kern1.Telu.TEL_ssatelugu.alt_L @kern2.Telu.TEL_rVocalicMatratelugu_R 180;
```

Comment:
the "ssa-telugu.alt"  glyph appears in both `@kern1.Default.TEL_ssatelugu.alt_L` group and in `@kern1.Telu.TEL_ssatelugu.alt_L` (it is the only member of the latter group);  the above kerning pair `pos @kern1.Telu.TEL_ssatelugu.alt_L @kern2.Telu.TEL_rVocalicMatratelugu_R 180;` looks to me useless duplication, because the pair immediately preceding already matches for "ssa-telugu.alt" and has even the same kern value (180). Does the source contain this unnecessary pair or is it us creating this?

Reply:
ssa-telugu.alt does not appear in `@kern1.Default.TEL_ssatelugu.alt_L`.